app/src/main/java/com/example/goodnote/ui/viewModels/NoteViewModel.kt

```diff
@@ -1,30 +1,93 @@
 package com.example.goodnote.ui.viewModels
 
-import android.content.Context
+import android.util.Log
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.example.goodnote.database.repository.NoteRepo
 import com.example.goodnote.database.models.Note
-import com.example.goodnote.utils.Injectors
+import com.example.goodnote.database.repository.NoteRepo
+import com.example.goodnote.utils.dummyNotes
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
 // AndroidViewModel(application) - use this one when you need application context
-// when do you need application context? repo/network only? or?
-class NoteViewModel(private val context: Context) : ViewModel() {
+// when do you need application context? repo/network only
+class NoteViewModel(private val repository: NoteRepo) : ViewModel() {
 
-    // inject repository here via viewmodelfactory, so the views are completely decoupled from repos
+    private val _notes: MutableLiveData<List<Note>> = MutableLiveData(dummyNotes)
+    val notes: LiveData<List<Note>>
+        get() = _notes
 
-    private val repository: NoteRepo = Injectors.getNoteRepository(context)
 
-    // dummy lists
-   private lateinit var _dummyNotes: MutableLiveData<List<Note>>
-    val dummyNotes: LiveData<List<Note>>
-        get() = _dummyNotes
+    private var _repoNotes: MutableLiveData<List<Note>> = MutableLiveData()
+    val repoNotes: LiveData<List<Note>>
+        get() = _repoNotes
+
+    init {
+        getAllNotes()
+    }
 
     // all repo functions here
-    fun saveNote(note: Note) = viewModelScope.launch {
+
+    fun getAllNotes() {
+         viewModelScope.launch(Dispatchers.IO) {
```

Comment:
`viewModelScope` by default launches new coroutine in default thread, which is totally fine. In this case, this thread is the android Main thread. :)

Here we could launch a coroutine without dispatcher parameter.

Example: 
```
   viewModelScope.launch {
      // here we do context switch to IO thread in order to get notes 
     val notes  = withContext(Dispetchers.IO) { repository.getAllNotes() }
     _repoNotes.value = notes
   }
```

As you can see above it is better to do thread switching only once by switching to IO thread when it is needed. :) 
This way we are saving processor time, since there won't be 2 context switches where one would be sufficient.